### ISSUE TITLE
Can't get serial data from EMR3 /EMR4 rs232

### ISSUE THREAD

**Issue by monotet** [bix6/smart-truck-public#1]
Hi! I have seen your project and noticed that it is related to what I'm trying to do.
I need to read data / flow rate from EMR3 (veeder-root) from rs232 to my Arduino uno but when I did code to get serial data. There are nothing showing on serial monitor.
Could you please suggest any way to get flow rate data by Arduino?

**Comment by bix6** [comment]
Hi @monotet! I don't see a repository on your profile so I can't see what you've tried so far. But if you take a look in the `experiments/` folder you may find a simple example of what you're looking for. If nothing is showing in the response, you're probably not sending a command it understands (it won't acknowledge a bad request, it'll just return nothing, iirc). Veeder Root has an emulator for PC you can use to send commands back/forth which may help you better understand the communication.

**Comment by monotet** [comment]
Just confused about your command below.If I want to get flow rate, where should I amend ? 

cmdSaleNum = b'\x7E\x01\xFF\x47\x73\x46\x7E'
cmdStart = b'\x7E\x01\xFF\x4F\x01\xB0\x7E'
cmdEnd = b'\x7E\x01\xFF\x4F\x03\xAE\x7E'
cmdLastTrans = b'\x7E\x01\xFF\x48\x01\x00\x00

**Comment by bix6** [comment]
You'll need to figure out the flow rate command from the documentation and convert it to the corresponding hex code commands as I did for those 4 (see `/docs/EMR3 docs/OBCSerialCommandsProtocol.PDF`). The emulator I mentioned can also help with this. Then you just need to send the command to the EMR3 and it should respond with the flow rate. You don't necessarily have to amend those other commands, just send the new one. Each of those lines is a command from the OBC Serial Commands Protocol PDF. I'm unsure if I broke that command down so I'm unsure if it's in any of my code.
